images/worker/Dockerfile

```diff
@@ -102,22 +102,25 @@ COPY pretend-bench.sh /usr/local/bin/bench
 COPY push_backup.py /usr/local/bin/push-backup
 COPY configure.py patched_bench_helper.py /usr/local/bin/
 COPY gevent_patch.py /opt/patches/
+COPY entrypoint.sh /usr/local/bin/entrypoint.sh
 
 WORKDIR /home/frappe/frappe-bench/sites
 
+ENTRYPOINT [ "entrypoint.sh" ]
+
 CMD [ "/home/frappe/frappe-bench/env/bin/gunicorn", "-b", "0.0.0.0:8000", "frappe.app:application" ]
 
 
 FROM configured_base as frappe
 
 RUN echo "frappe" >/home/frappe/frappe-bench/sites/apps.txt
-COPY --from=frappe_builder /home/frappe/frappe-bench/apps/frappe /home/frappe/frappe-bench/apps/frappe
-COPY --from=frappe_builder /home/frappe/frappe-bench/env /home/frappe/frappe-bench/env
+COPY --from=frappe_builder --chown=frappe:frappe /home/frappe/frappe-bench/apps/frappe /home/frappe/frappe-bench/apps/frappe
```

Comment:
Better not to allow non-root users to change code. Security.

Reply:
this causes permission denied for `ln` command

Reply:
I'll prefer frappe and other apps be installed by user frappe and not root. This was what was done in previous image.

It may be anti-pattern but it make patching easy. For depending images as well as runtime.

Reply:
> It may be anti-pattern but it make patching easy.

In images based on Frappe it is easy to switch to root and back to frappe user. If you think it is the best to give rights to frappe user, let it be so.